RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -118,16 +116,16 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const ChamberHitContainer& u
     }
 
     //change the tresholds if it's time to look for displaced mu segments
-    if(doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
-      doCollisions = false;
+    if(l_doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
```

Comment:
I'm missing the point though: what is wrong with the syntax or the expression?
if size() is smaller than used_rh, it will still produce a correct signed answer, at least with the normal range of values (both sufficiently below 2^31-1).

Please note that the type of size() is 64-bit unsigned.
So, in a more abstract case, where size() is larger than 2^31-1, the proposed change will more likely produce an incorrect result